trieste/bayesian_optimizer.py

```diff
@@ -444,25 +445,33 @@ def optimize(
         models = cast(
             Dict[str, TrainableProbabilisticModelType], map_values(create_model, model_specs)
         )
+
         history: list[Record[StateType]] = []
 
         for step in range(num_steps):
             set_step_number(step)
             try:
-                if track_state:
-                    models_copy = copy.deepcopy(models)
-                    acquisition_state_copy = copy.deepcopy(acquisition_state)
-                    history.append(Record(datasets, models_copy, acquisition_state_copy))
+
+                t_0_total = time()  # start wallclock for total BO step time
```

Comment:
Use time.perf_counter instead as it's (1) more accurate and (2) doesn't depend on the system clock so can't go backwards if somebody e.g. changes the system to summer time during execution!

```suggestion
                t_0_total = perf_counter()  # start wallclock for total BO step time
```